lib/rucio/core/replica.py

```diff
@@ -2745,138 +2991,153 @@ def list_dataset_replicas(scope, name, deep=False, *, session: "Session"):
-                              models.CollectionReplica.bytes,
-                              models.CollectionReplica.length,
-                              models.CollectionReplica.available_bytes,
-                              models.CollectionReplica.available_replicas_cnt.label("available_length"),
-                              models.CollectionReplica.state,
-                              models.CollectionReplica.created_at,
-                              models.CollectionReplica.updated_at,
-                              models.CollectionReplica.accessed_at)\
-            .filter_by(scope=scope, name=name, did_type=DIDType.DATASET)\
-            .filter(models.CollectionReplica.rse_id == models.RSE.id)\
-            .filter(models.RSE.deleted == false())
-
-        for row in query:
+        stmt = select(
+            models.CollectionReplica.scope,
+            models.CollectionReplica.name,
+            models.RSE.rse,
+            models.CollectionReplica.rse_id,
+            models.CollectionReplica.bytes,
+            models.CollectionReplica.length,
+            models.CollectionReplica.available_bytes,
+            models.CollectionReplica.available_replicas_cnt.label("available_length"),
+            models.CollectionReplica.state,
+            models.CollectionReplica.created_at,
+            models.CollectionReplica.updated_at,
+            models.CollectionReplica.accessed_at
+        ).where(
+            and_(models.CollectionReplica.scope == scope,
+                 models.CollectionReplica.name == name,
+                 models.CollectionReplica.did_type == DIDType.DATASET,
+                 models.CollectionReplica.rse_id == models.RSE.id,
+                 models.RSE.deleted == false())
+        )
+
+        for row in session.execute(stmt).all():
             yield row._asdict()
 
     else:
 
-        # find maximum values
-        content_query = session\
-            .query(func.sum(models.DataIdentifierAssociation.bytes).label("bytes"),
-                   func.count().label("length"))\
-            .with_hint(models.DataIdentifierAssociation, "INDEX_RS_ASC(CONTENTS CONTENTS_PK) NO_INDEX_FFS(CONTENTS CONTENTS_PK)", 'oracle')\
-            .filter(models.DataIdentifierAssociation.scope == scope)\
-            .filter(models.DataIdentifierAssociation.name == name)
+        stmt = select(
+            func.sum(models.DataIdentifierAssociation.bytes).label("bytes"),
+            func.count().label("length")
+        ).select_from(
+            models.DataIdentifierAssociation
+        ).with_hint(
+            models.DataIdentifierAssociation,
+            'INDEX_RS_ASC(CONTENTS CONTENTS_PK) NO_INDEX_FFS(CONTENTS CONTENTS_PK)', 'oracle'
+        ).where(
+            and_(models.DataIdentifierAssociation.scope == scope,
+                 models.DataIdentifierAssociation.name == name)
+        )
 
         bytes_, length = 0, 0
-        for row in content_query:
+        for row in session.execute(stmt).all():
```

Comment:
I don't think we need a loop here actually, what do you think about the following change?

From:
```
        bytes_, length = 0, 0
        for row in content_query:
            bytes_, length = row.bytes, row.length
 ```
 
 To:
 ```
         bytes_, length = session.execute(stmt).one()
         bytes_ = bytes_ or 0
 ```
 
 The previous query always only returned a single row, and set `bytes_` and `length` based on this. The reason I included `bytes_ = bytes_ or 0` is because `func.sum().label('bytes')` might return null, whilst `func.count().label('length') `cannot. This also means that .one() will never raise an exception as .count() always returns an integer, so if no values match the query it returns [NULL, 0]

Reply:
If it works it works! No problem with that